backend/tarsy/controllers/auth.py

```
@@ -0,0 +1,264 @@
+        # Validate OAuth CSRF token exists in database
+        oauth_state = oauth_repo.get_state(csrf_token)
+        if not oauth_state or oauth_state.expires_at < now_us():
+            raise HTTPException(400, "Invalid or expired OAuth state")
+        
+        # Clean up used state
+        oauth_repo.delete_state(csrf_token)
+        
+        # Exchange code for access token using requests-oauthlib
+        from requests_oauthlib import OAuth2Session
+        
+        logger.debug(f"Exchanging OAuth code for access token using requests-oauthlib")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)